lib/topology_openswitch/vtysh.py

```diff
@@ -170,9 +161,19 @@ def _determine_set_prompt(self, connection=None):
         # always return the produced output, even if an EOF exception
         # follows after it. This is done to handle the segmentation fault
         # errors.
-        spawn.sendline('stdbuf -oL vtysh')
-        spawn.expect(VTYSH_STANDARD_PROMPT)
-
+        vtysh_connected = False
+        for i in range(10):
+            try:
+                spawn.sendline('stdbuf -oL vtysh')
+                spawn.expect(VTYSH_STANDARD_PROMPT, timeout=3)
+                vtysh_connected = True
+                break
+            except:
+                continue
+        if not vtysh_connected:
+            raise Exception(
+                'Unable to connect to vtysh, before: {}'
+                .format(spawn.before.decode('utf-8', errors='ignore')))
```

Comment:
That is a matter of personal taste. I myself prefer to refrain from creating single-use variables.